File: Practica5/java/Modular-Estructurada/Ecuacioncuadratica.java
import java.util.Scanner;

public class Ecuacioncuadratica {
    public static float getDiscriminante(float a, float b, float c) {
        return (b*b) - (4*a*c);
    }
    public static Float getRaiz1(float a, float b, float discriminante) {
        if (discriminante<0) {
        	return null;
        }
        return (-b +(float) Math.sqrt(discriminante))/(2*a);
    }
    public static Float getRaiz2(float a, float b, float discriminante) {
        if (discriminante<0) {
        	return null;       
        }
        return (-b-(float) Math.sqrt(discriminante))/(2*a);
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese los valores de a, b y c: ");
        float a = scanner.nextFloat();
        float b = scanner.nextFloat();
        float c = scanner.nextFloat();
        float discriminante = getDiscriminante(a, b, c);
        Float raiz1 = getRaiz1(a, b, discriminante);
        Float raiz2 = getRaiz2(a, b, discriminante);
        if (raiz1 == null) {
            System.out.println("La ecuación no tiene raíces reales.");
        } else if (raiz1.equals(raiz2)) {
            System.out.printf("La ecuación tiene una raíz: %.5f%n", raiz1);
        } else {
            System.out.printf("La ecuación tiene dos raíces: %.5f y %.5f%n", raiz1, raiz2);
        }
        scanner.close();
    }
}

File: Practica5/Python/Modular-Estructurada/Ecuacioncuadratica.py
import math

def getDiscriminante(a, b, c):
    return b**2 - 4*a*c
def getRaiz1(a, b, discriminante):
    return (-b + math.sqrt(discriminante)) / (2*a)

def getRaiz2(a, b, discriminante):
    return (-b - math.sqrt(discriminante)) / (2*a)
def resolver_ecuacion(a, b, c):
    discriminante = getDiscriminante(a, b, c)   
    if discriminante > 0:
        r1 = getRaiz1(a, b, discriminante)
        r2 = getRaiz2(a, b, discriminante)
        print(f"La ecuación tiene dos raíces {r1:.6f} y {r2:.6f}")
    elif discriminante == 0:
        r1 = -b / (2*a)
        print(f"La ecuación tiene una raíz {r1:.6f}")
    else:
        print("La ecuación no tiene raíces reales")
if __name__ == "__main__":
    a, b, c = map(float, input("Ingrese a, b, c: ").split())
    resolver_ecuacion(a, b, c)
1.0